src/Tasks/AssemblyDependency/ReferenceTable.cs

```diff
@@ -2667,12 +2667,12 @@ internal void GetReferenceItems
                 }
             }
 
-            primaryFiles = primaryItems.ToArray();
-            dependencyFiles = dependencyItems.ToArray();
-            relatedFiles = relatedItems.ToArray();
-            satelliteFiles = satelliteItems.ToArray();
-            serializationAssemblyFiles = serializationAssemblyItems.ToArray();
-            scatterFiles = scatterItems.ToArray();
+            primaryFiles = NormalizePathsAsArray(primaryItems);
+            dependencyFiles = NormalizePathsAsArray(dependencyItems);
+            relatedFiles = NormalizePathsAsArray(relatedItems);
+            satelliteFiles = NormalizePathsAsArray(satelliteItems);
+            serializationAssemblyFiles = NormalizePathsAsArray(serializationAssemblyItems);
+            scatterFiles = NormalizePathsAsArray(scatterItems);
 
              // Sort for stable outputs. (These came from a dictionary, which has undefined enumeration order.)
             Array.Sort(primaryFiles, TaskItemSpecFilenameComparer.GenericComparer);
```

Comment:
Should we be concerned that we're changing the sort order here?

Reply:
From the comment, I don't _think_ so. The comment and the sort have been there since the initial (GitHub) code commit, but it sounds like it's important that if you run RAR twice, the order is the same, but not so important what that order actually is. This maintains that there is a correct ordering, albeit reordering it slightly.